tests/ingest_runs/test_genomic_data_loader.py

```diff
@@ -0,0 +1,396 @@
+        CONCEPT.BIOSPECIMEN.TARGET_SERVICE_ID,
+        CONCEPT.GENOMIC_FILE.DATA_TYPE,
+        CONCEPT.GENOMIC_FILE.FILE_NAME,
+        CONCEPT.GENOMIC_FILE.HASH_DICT,
+        CONCEPT.GENOMIC_FILE.SIZE,
+        CONCEPT.GENOMIC_FILE.URL_LIST,
+        CONCEPT.GENOMIC_FILE.FILE_FORMAT,
+        CONCEPT.GENOMIC_FILE.ID,
+        CONCEPT.GENOMIC_FILE.SOURCE_FILE,
+        CONCEPT.GENOMIC_FILE.HARMONIZED,
+        CONCEPT.GENOMIC_FILE.VISIBLE,
+    }
+    for column in expected_columns:
+        assert column in df
+    assert df.shape[1] == len(expected_columns)
+
+    mock_load_bs_gf_links.assert_called_once()
+    load_bs_arg_df = mock_load_bs_gf_links.call_args_list[0][0][0]
+    load_bs_arg_df = load_bs_arg_df.drop_duplicates(
+        [CONCEPT.BIOSPECIMEN.TARGET_SERVICE_ID]
+    )
+    assert type(load_bs_arg_df) == pd.DataFrame
+    assert not load_bs_arg_df.empty
+    assert load_bs_arg_df.shape[0] == df.shape[0]
+    expected_columns.add(CONCEPT.GENOMIC_FILE.TARGET_SERVICE_ID)
+    for column in expected_columns:
+        assert column in load_bs_arg_df
+    assert load_bs_arg_df.shape[1] == len(expected_columns)
+    mock_load_se_gf_links.assert_called_once()
+
+
+def test_load_specimen_harmonized_gf_links(
+    study_generator, mock_s3_scrape, mock_load_entities, mock_load_se_gf_links,
+    mock_validate
+):
+    """
+    Test the load_specimen_harmonized_gf_links function. This is the second
+    step of the ingestion process. Mock the steps after.
+    """
+    loader = GenomicDataLoader(
+        study_id=study_generator.study_id, init_logger=True
+    )
+    manifest_df = study_generator.dataframes['gwo_manifest.tsv']
+    loader.ingest_gwo(manifest_df)
+    mock_s3_scrape.assert_called_once()
+    assert mock_load_entities.call_count == 2
+    # We already checked that the first call to mock_load_entities had the
+    # appropriate arguments in the previous test, and they won't change. So
+    # we can just do rudimentary checks here.
+    entity_type1, df1 = mock_load_entities.call_args_list[0][0]
+    assert entity_type1 == "genomic_file"
+    assert type(df1) == pd.DataFrame
+    assert not df1.empty
+    manifest_df = manifest_df.drop_duplicates(["KF Participant ID"])
+    assert manifest_df.shape[0] == df1.shape[0]
+
+    # Now check the biospecimen gf links loading
+    entity_type2, df2 = mock_load_entities.call_args_list[1][0]
+    assert entity_type2 == "biospecimen_genomic_file"
+    assert type(df2) == pd.DataFrame
+    assert not df2.empty
+    assert manifest_df.shape[0] == df2.shape[0]
+    expected_columns = {
+        CONCEPT.GENOMIC_FILE.ID,
+        CONCEPT.BIOSPECIMEN.TARGET_SERVICE_ID,
+        CONCEPT.GENOMIC_FILE.TARGET_SERVICE_ID,
+        CONCEPT.BIOSPECIMEN.ID,
+        CONCEPT.BIOSPECIMEN_GENOMIC_FILE.VISIBLE,
+    }
+    for col in expected_columns:
+        assert col in df2
+    assert df2.shape[1] == len(expected_columns)
+
+    # Finally check load_seq_exp_harmonized_genomic_files was called with the
```

Comment:
Similar to a previous comment. This stuff should probably in its own test for the `load_seq_exp_harmonized_genomic_files` method where the test setup would simply be:

```
def test_load_seq_exp_harmonized_genomic_files(
    study_generator, mock_s3_scrape, mock_load_entities, 
):
    loader = GenomicDataLoader(
        study_id=study_generator.study_id, init_logger=True
    )
    manifest_df = study_generator.dataframes['gwo_manifest.tsv']
    df = loader.ingest_gwo(manifest_df)
    # All the checks you have below this line would be heree
```